CharlieBackend.Business/Services/ScheduleServiceFolder/EventsService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Business.Services.ScheduleServiceFolder.Helpers;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+
+namespace CharlieBackend.Business.Services.ScheduleServiceFolder
+{
+    public class EventsService : IEventsService
+    {
+
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+        private readonly IScheduledEventHandlerFactory _scheduledEventFactory;
+        private readonly SchedulesEventsValidator _validator;
+
+        public EventsService(IUnitOfWork unitOfWork, IMapper mapper, IScheduledEventHandlerFactory scheduledEventHandlerFactory)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+            _scheduledEventFactory = scheduledEventHandlerFactory;
+            _validator = new SchedulesEventsValidator(unitOfWork);
+        }
+
+        public async Task<Result<bool>> DeleteConcreteScheduleByIdAsync(long id)
+        {
+            if (id < 0)
+            {
+                return Result<bool>.GetError(ErrorCode.Conflict,
+                    "Can not delete scheduled event due to wrong request data");
+            }
+
+            var scheduledEvent = await _unitOfWork.ScheduledEventRepository.GetByIdAsync(id);
+
+            if (scheduledEvent is null)
+            {
+                return Result<bool>.GetError(ErrorCode.ValidationError, "Scheduled event does not exist");
+            }
+
+            await _unitOfWork.ScheduledEventRepository.DeleteAsync(id);
+            await _unitOfWork.CommitAsync();
+
+            return Result<bool>.GetSuccess(true);
+        }
+
+        public async Task<Result<ScheduledEventDTO>> UpdateScheduledEventByID(long scheduledEventId, UpdateScheduledEventDto request)
+        {
+            string error = await _validator.ValidateScheduledEventId(scheduledEventId);
+
+            error ??= await _validator.ValidateUpdateScheduleDTO(request);
+
+            if (error != null)
+            {
+                return Result<ScheduledEventDTO>.GetError(ErrorCode.ValidationError, error);
```

Comment:
Create new exception. You can call this exception EventValidationExceptoin. Add this exception to ErrorHandlerMiddleware to a switch case to return 400 responses. Throw EventValidationExceptoin here with an error message.

Reply:
Done

Reply:
But I used a switch with checking type and name for the exception and then comparing it with the name of the exception. Unfortunately, C# 9.0 only is able to check exceptions inside its body